### DISCUSSION TITLE
Safe boot with esp32

### DISCUSSION THREAD

**Discussion by yobio** [micropython/micropython#9798] [Hardware & Peripherals]
Hi,
I have an esp32 board (ttgo t-call) running MicroPython. I would like to use the UART connection on the USB port but it is already used by the REPL. So I want to disable the UART REPL but still be able to access it easily (WebREPL could cause connection errors).
I saw that there is a safe mode on the WiPy card to temporarily cancel the execution of boot.py by setting the GP28 to high during boot. So I could access the UART REPL by activating this pin. However I didn't find anything about this for esp32. Is this possible?
If not, would you have another solution for me?

I'm using this firmware: [v1.19.1 (2022-06-18)](https://micropython.org/download/esp32spiram/).

Thanks for your help!

**Comment by ThinkTransit** [comment]
If you disable the USB UART you can still access the repl through the pins using a usb-serial, is that a option?

**Reply by yobio** [reply]
Thank you for your answer. Sorry if I wasn't very clear in my question. I'd like to disable the usb repl to be able to use the usb uart (UART0) without interference with the repl. There are two other uart interfaces but I can't change the pins of the uart repl without editing the firmware source code.
